addons/l10n_sa_edi/models/account_move.py

```diff
@@ -190,3 +190,17 @@ def _l10n_sa_is_in_chain(self):
         """
         zatca_doc_ids = self.edi_document_ids.filtered(lambda d: d.edi_format_id.code == 'sa_zatca')
         return len(zatca_doc_ids) > 0 and not any(zatca_doc_ids.filtered(lambda d: d.state == 'to_send'))
+
+
+class AccountMoveLine(models.Model):
+    _inherit = 'account.move.line'
+
+    @api.depends('price_subtotal', 'price_total')
+    def _compute_tax_amount(self):
+        super()._compute_tax_amount()
+        for record in self:
+            move = record.move_id
+            if move.country_code == 'SA':
+                taxes_vals = move._prepare_edi_tax_details(filter_to_apply=self.env['account.edi.xml.ubl_21.zatca']._apply_invoice_tax_filter,
```

Comment:
Hmm, we could group by move, so we only calculate the tax details once for every move.  There might be an issue for downpayment lines however.  (as this thing filters them out for the XML, but here, we might need to show it instead of 0 numbers. ) @kitan191

Reply:
@mebe-odoo please check as well.

Reply:
@jco-odoo Thanks for the review.

I've updated the fix to only compute the tax details once per move.
I've also removed the filter made on the down-payment lines.